v2/internal/testcommon/samples_tester.go

```diff
@@ -228,3 +235,43 @@ func IsSampleExcluded(path string, exclusions []string) bool {
 	}
 	return false
 }
+
+func (t *SamplesTester) findAndSetARMIDReferences(obj genruntime.ARMMetaObject) (genruntime.ARMMetaObject, error) {
```

Comment:
Improve this method name? 

"And" in a method name is _almost_ universally a code smell, but here, the method is genuinely doing one thing so the fix is a better name.

Maybe `updateARMReferencesForTest` ?

It definitely needs a docComment to explain what's being updated.